### ISSUE TITLE
Generated upstream locations have non-deterministic order: arbitrary reloads of nginx

### ISSUE THREAD

**Issue by languitar** [nginx-proxy/nginx-proxy#1144]
We have a setup where nginx-unrelated containers are spawned frequently. However, each change to the list of running container seems to trigger a regeneration of the configuration files via docker-gen. What we see is that sometimes nginx gets reloaded, even though nothing nginx-related has changed. We could boil this down to the generation of the server list in the `upstream` section of the nginx config file. The ordering of the networks somewhere around the template line `{{ range $knownNetwork := $CurrentContainer.Networks }}` seems to be non-deterministic and thus the upstream configurations for our containers having multiple networks get permuted randomly. For instance, we can see such diffs when comparing revisions of the generated config file:

```
--- /tmp/default.conf   2018-06-21 14:09:39.145917826 +0000
+++ default.conf        2018-06-21 14:28:08.926412830 +0000
@@ -88,11 +88,11 @@
 }
 # XXX
 upstream XXX {
-                               # Cannot connect to network of this container
-                               server 127.0.0.1 down;
                                ## Can be connected with "nginx-proxy" network
                        # gitea-cse-server
                        server 172.23.0.4:3000;
+                               # Cannot connect to network of this container
+                               server 127.0.0.1 down;
 }
 server {
        server_name XXX;
```

Thus, the network list has to be sorted before generating the config file to prevent unrelated webserver reloads.

**Comment by jpic** [comment]
Do you think #1302 would also fix the outcome of this issue ?

**Comment by languitar** [comment]
> Do you think #1302 would also fix the root cause of your issue ?

I don't know if there are more things that could get reordered randomly. In the case that I posted that would probably be enough.

> Why do you think you need server 127.0.0.1 down in upstream ?

We didn't explicitly enable those lines?

**Comment by jpic** [comment]
> We didn't explicitly enable those lines?

I think they got there by mistake, I don't understand the purpose of these lines which don't make sense to me, and cause quite some problems like, nginx not starting.

**Comment by buchdag** [comment]
Fixed by #1985
